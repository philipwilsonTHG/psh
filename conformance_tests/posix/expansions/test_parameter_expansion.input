# Test parameter expansion features

# Setup test variables
var="hello"
empty=""
unset unset_var

# Test basic expansion
echo "Basic: $var"
echo "Braced: ${var}"

# Test default value expansions
echo "Default if unset: ${unset_var:-default}"
echo "Default if empty: ${empty:-default}"
echo "Set value: ${var:-default}"

# Test alternative expansions
echo "Alt if set: ${var:+alternative}"
echo "Alt if unset: ${unset_var:+alternative}"

# Test without colon (null/unset distinction)
empty=""
echo "Without colon - empty: ${empty-default}"
echo "With colon - empty: ${empty:-default}"

# Test string length
echo "Length of var: ${#var}"
echo "Length of empty: ${#empty}"

# Test with special variables
set -- arg1 arg2 arg3
echo "Length of all args: ${#*}"
echo "Length of \$1: ${#1}"