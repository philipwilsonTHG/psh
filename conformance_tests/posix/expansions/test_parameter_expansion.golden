Basic: hello
Braced: hello
Default if unset: default
Default if empty: default
Set value: hello
Alt if set: alternative
Alt if unset:
Without colon - empty:
With colon - empty: default
Length of var: 5
Length of empty: 0
Length of all args: 14
Length of $1: 4
